### PR TITLE
Moved `ScaffolderEntitiesProcessor` and `catalogModuleTemplateKind` to a dedicated module

### PR DESCRIPTION
**Pull request by freben**

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "@backstage/plugin-azure-sites-backend": "workspace:^",
     "@backstage/plugin-badges-backend": "workspace:^",
     "@backstage/plugin-catalog-backend": "workspace:^",
+    "@backstage/plugin-catalog-backend-module-scaffolder-entities": "^0.0.0",
```
Comment (benjdlambert) [inline packages/backend/package.json:43]: `workspace:^`?

File: packages/backend-next/package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "@backstage/plugin-azure-devops-backend": "workspace:^",
     "@backstage/plugin-badges-backend": "workspace:^",
     "@backstage/plugin-catalog-backend": "workspace:^",
+    "@backstage/plugin-catalog-backend-module-scaffolder-entities": "^0.0.0",
```
Comment (benjdlambert) [inline packages/backend-next/package.json:37]: `workspace:^`?

File: plugins/catalog-backend-module-scaffolder-entities/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "@backstage/plugin-catalog-backend-module-scaffolder-entities",
```
Comment (benjdlambert) [inline plugins/catalog-backend-module-scaffolder-entities/package.json:2]: Hmm, trying to think of a better name here, but I guess it's fine for a catch all if we decide to add some other things other than just the `kind: Template` stuff here too.

**Review by benjdlambert — CHANGES_REQUESTED** [review]
NIce :tada: just the workspace stuff

File: packages/backend/package.json (hunk 1 above)
Comment (freben) [reply packages/backend/package.json]: Oh wut, how did I manage to get this to be a semver range

Fix'd

File: plugins/catalog-backend-module-scaffolder-entities/package.json (hunk 3 above)
Comment (freben) [reply plugins/catalog-backend-module-scaffolder-entities/package.json:2]: Yeah we pondered it, started out with `-template-kind` but given what the name of the processor was, we kinda shifted to the more generic naming. Funnily though, the module is still named `catalogModuleTemplateKind`, and that's supposedly the future ... :) Should THAT one be renamed to `catalogModuleScaffolderEntityModel` or something like that?

**Pull request merged**